pkg/apply/task/inv_set_task.go

```diff
@@ -54,79 +57,114 @@ func (i *InvSetTask) Identifiers() object.ObjMetadataSet {
+
+	// If an object failed to delete and was previously stored in the inventory,
+	// then keep it in the inventory so it can be applied/pruned next time.
+	// It's likely that all the delete failures are already in the inventory,
+	// because the set of resources to prune comes from the inventory,
+	// but we're doing the intersection anyway just to be sure.
+	pruneFailures := i.PrevInventory.Intersection(im.FailedDeletes())
+	klog.V(4).Infof("set inventory %d failed prunes", len(pruneFailures))
+	invObjs = invObjs.Union(pruneFailures)
+
+	// If an object skipped delete and was previously stored in the inventory,
+	// then keep it in the inventory so it can be applied/pruned next time.
+	// It's likely that all the skipped deletes are already in the inventory,
+	// because the set of resources to prune comes from the inventory,
+	// but we're doing the intersection anyway just to be sure.
+	pruneSkips := i.PrevInventory.Intersection(im.SkippedDeletes())
+	klog.V(4).Infof("keep in inventory %d skipped prunes", len(pruneSkips))
+	invObjs = invObjs.Union(pruneSkips)
+
+	// If an object is abandoned, then remove it from the inventory.
+	abandonedObjects := taskContext.AbandonedObjects()
+	klog.V(4).Infof("remove from inventory %d abandoned objects", len(abandonedObjects))
+	invObjs = invObjs.Diff(abandonedObjects)
+
+	// If an object is invalid and was previously stored in the inventory,
+	// then keep it in the inventory so it can be applied/pruned next time.
+	invalidObjects := i.PrevInventory.Intersection(taskContext.InvalidObjects())
+	klog.V(4).Infof("keep in inventory %d invalid objects", len(invalidObjects))
+	invObjs = invObjs.Union(invalidObjects)
+
+	klog.V(4).Infof("get the apply status for %d objects", len(invObjs))
+	objStatus := taskContext.InventoryManager().Inventory().Status.Objects
+
+	klog.V(4).Infof("set inventory %d total objects", len(invObjs))
+	err := i.InvClient.Replace(i.InvInfo, invObjs, objStatus, i.DryRun)
+
+	klog.V(2).Infof("inventory set task completing (name: %q)", i.TaskName)
+	return err
+}
+
+// deleteInventory deletes the inventory object from the cluster.
+func (i *DeleteOrUpdateInvTask) deleteInventory() error {
+	klog.V(2).Infof("delete inventory task starting (name: %q)", i.Name())
+	err := i.InvClient.DeleteInventoryObj(i.InvInfo, i.DryRun)
+	// Not found is not error, since this means it was already deleted.
+	if apierrors.IsNotFound(err) {
+		err = nil
+	}
+	klog.V(2).Infof("delete inventory task completing (name: %q)", i.Name())
+	return err
+}
+
+// Start will reconcile the state of the inventory, depending on the value of
+// Destroy.
+//
+// If Destroy is set, the intent is to delete the inventory. The inventory will
+// only be deleted if all prunes were successful (none failed/skipped). If any
+// prunes were failed or skipped, the inventory will be updated.
+//
+// If Destroy is false, the inventory will be updated.
+func (i *DeleteOrUpdateInvTask) Start(taskContext *taskrunner.TaskContext) {
```

Comment:
i usually like to put public methods above private methods, since it's more likely someone came to the file to find them. interface before impl.

Reply:
done